torch_geometric/datasets/lrgb.py

```diff
@@ -0,0 +1,330 @@
+              - Node Prediction
+              - 123,286
+              - ~476.88
+              - ~2,693.67
+              - 81
+            * - PCQM-Contact
+              - Link Prediction
+              - 529,434
+              - ~30.14
+              - ~61.09
+              - 1
+            * - Peptides-func
+              - Graph Classification
+              - 15,535
+              - ~150.94
+              - ~307.30
+              - 10
+            * - Peptides-struct
+              - Graph Regression
+              - 15,535
+              - ~150.94
+              - ~307.30
+              - 11
+
+    """
+
+    names = [
+        'pascalvoc-sp', 'coco-sp', 'pcqm-contact', 'peptides-func',
+        'peptides-struct'
+    ]
+
+    urls = {
+        'pascalvoc-sp':
+        'https://www.dropbox.com/s/8x722ai272wqwl4/pascalvocsp.zip?dl=1',
+        'coco-sp':
+        'https://www.dropbox.com/s/r6ihg1f4pmyjjy0/cocosp.zip?dl=1',
+        'pcqm-contact':
+        'https://www.dropbox.com/s/qdag867u6h6i60y/pcqmcontact.zip?dl=1',
+        'peptides-func':
+        'https://www.dropbox.com/s/ycsq37q8sxs1ou8/peptidesfunc.zip?dl=1',
+        'peptides-struct':
+        'https://www.dropbox.com/s/zgv4z8fcpmknhs8/peptidesstruct.zip?dl=1'
+    }
+
+    dwnld_file_name = {
+        'pascalvoc-sp': 'voc_superpixels_edge_wt_region_boundary',
+        'coco-sp': 'coco_superpixels_edge_wt_region_boundary',
+        'pcqm-contact': 'pcqmcontact',
+        'peptides-func': 'peptidesfunc',
+        'peptides-struct': 'peptidesstruct'
+    }
+
+    def __init__(self, root: str, name: str, split: str = "train",
+                 transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None,
+                 pre_filter: Optional[Callable] = None):
+        self.name = name.lower()
+        assert self.name in self.names
+        assert split in ['train', 'val', 'test']
+
+        super().__init__(root, transform, pre_transform, pre_filter)
+        path = osp.join(self.processed_dir, f'{split}.pt')
+        self.data, self.slices = torch.load(path)
+
+    @property
+    def raw_dir(self) -> str:
+        return osp.join(self.root, self.name, 'raw')
+
+    @property
+    def processed_dir(self) -> str:
+        return osp.join(self.root, self.name, 'processed')
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        if self.name.split('-')[1] == 'sp':
+            return ['train.pickle', 'val.pickle', 'test.pickle']
+        else:
+            return ['train.pt', 'val.pt', 'test.pt']
+
+    @property
+    def processed_file_names(self) -> List[str]:
+        return ['train.pt', 'val.pt', 'test.pt']
+
+    def download(self):
+        shutil.rmtree(self.raw_dir)
+        path = download_url(self.urls[self.name], self.root)
+        extract_zip(path, self.root)
+        os.rename(osp.join(self.root, self.dwnld_file_name[self.name]),
+                  self.raw_dir)
+        os.unlink(path)
+
+    def process(self):
```

Comment:
Since `process_sp`  and `process_peptides` work in a similar way, can we unify them into one single processing function?